src/libmongoc/src/mongoc/mongoc-apm.c

```diff
@@ -136,12 +191,23 @@ mongoc_apm_command_succeeded_init (mongoc_apm_command_succeeded_t *event,
                                    int64_t operation_id,
                                    const mongoc_host_list_t *host,
                                    uint32_t server_id,
+                                   bool force_redaction,
```

Comment:
To check, is the rationale for `is_redacted` and `force_redaction` to avoid a needless call to `mongoc_apm_is_sensitive_reply`? I.e. if we know the command is redacted, we can redact the reply without having to inspect it.

Reply:
That wasn't the implicit goal, but it's a nice side-effect. This was required to be able to redact replies to a `hello` command with speculative authentication even when the reply doesn't contain sensitive information (i.e., no `speculativeAuthenticate` field).